src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -85,10 +80,54 @@ winrt::Microsoft::Terminal::Settings::Model::CascadiaSettings CascadiaSettings::
     settings->_userSettingsString = _userSettingsString;
     settings->_userSettings = _userSettings;
     settings->_defaultSettings = _defaultSettings;
-    settings->_userDefaultProfileSettings = _userDefaultProfileSettings;
+
+    _CopyProfileInheritanceTree(settings.get());
+
     return *settings;
 }
 
+// Method Description:
+// - Copies the inheritance tree for profiles and hooks them up to a clone CascadiaSettings
+// Arguments:
+// - cloneSettings: the CascadiaSettings we're copying the inheritance tree to
+// Return Value:
+// - <none>
+void CascadiaSettings::_CopyProfileInheritanceTree(CascadiaSettings* cloneSettings) const

```

Comment:
Is there a reason this is a `CascadiaSettings*`? That seems dangerous to me. Shouldn't that be like, a `com_ptr<>&` or something?